src/mesh/boundary_standard.cxx

```diff
@@ -2167,6 +2201,10 @@ void BoundaryNeumann_O4::apply(Field2D& f, BoutReal t) {
       }
     }
   }
+#else
+  throw BoutException(
+      "Applying boundary to Field2D not compatible with 3D metrics in all cases.");
```

Comment:
2 things:  
1.) should this be fixed before the merge? I feel like it can break some code, and cause models to be written specifically for 2D/3D, even though we might want to be able to switch. 
2.) Should this be less general (like, for instance, line 2309)

Reply:
Does anybody even use the code? I have my doubts that codes use Field2D a lot. And those few that use Field2D probably have to switch to 3D anyway, (e.g. auto or Coordinates::metric_field_type) or keep the 3D metric disabled until we rewrite as variant

Reply:
test-laplacexy uses it.  (and Hermes uses laplacexy)